modules/CMakeLists.txt

```diff
@@ -44,6 +39,17 @@ FetchContent_Declare(
 )
 FetchContent_MakeAvailable(SPIRV-Headers)
 
+if(CA_NATIVE_CPU)
```

Comment:
`FetchContent` needs to be called before adding `mux` and `compiler`, but I'd like to avoid adding `debug`, `loader` and `utils` for native cpu. I've refactored my changes so that the intent is more clear (hopefully)